src/geopackage/sqlite/database.cpp

```diff
@@ -0,0 +1,108 @@
+#include "NGen_SQLite.hpp"
+
+using namespace geopackage;
+
+/**
+ * Get a runtime error based on a function and code.
+ *
+ * @param f String denoting the function where the error originated
+ * @param code sqlite3 result code
+ * @param extra additional messages to add to the end of the error
+ * @return std::runtime_error
+ */
+std::runtime_error sqlite_error(const std::string& f, int code, const std::string& extra = "")
+{
+    std::string errmsg = f + " returned code "
+                           + std::to_string(code)
+                           + " (msg: "
+                           + std::string(sqlite3_errstr(code))
+                           + ")";
+
+    if (!extra.empty()) {
+      errmsg += " ";
+      errmsg += extra;
+    }
+
+    return std::runtime_error(errmsg);
+}
+
+sqlite::sqlite(const std::string& path)
+{
+    sqlite3* conn;
```

Comment:
Tiny remark: It's not a great idea to name a local variable to shadow a member variable. That is a case where qualifying `this->conn` makes a difference, though.

Reply:
I may have a tendency to do this (on accident) :) but I guess it's a good thing I'm on the side of prefacing all members with `this->` 😛